File: Dante1/Module1.07/Zadanie7.08/main.c
#include <stdio.h>

int compare(int, int, int);

int main()
{
    int num1, num2, num3;

    printf("Enter numbers: ");

    int check = scanf("%i-%i-%i",&num1,&num2,&num3);
    if(check != 3)
    {
        printf("Incorrect input\n");
        return 1;
    }

    if(compare(num1, num2, num3) == 1)
    {
        printf("EQUAL");
    }
    else
    {
        printf("NOT EQUAL");
    }

    return 0;
}

int compare(int a, int b, int c)
{
    if(a == b && b == c)
    {
        return 1;
    }
    return 0;
}

File: Dante1/Module1.07/Zadanie7.08/main.rs
fn compare(num1: i32, num2: i32, num3: i32) -> i32 {
    if num1 == num2 && num2 == num3 {
        return 1;
    }
    0
}

fn main() {
    let mut input: String  = String::new();
    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("err");

    let mut numbers = input.trim().split_whitespace();

    let num1: i32 = numbers.next().expect("Нет первого числа").parse().expect("Ошибка парсинга");
    let num2: i32 = numbers.next().expect("Нет второго числа").parse().expect("Ошибка парсинга");
    let num3: i32 = numbers.next().expect("Нет третьего числа").parse().expect("Ошибка парсинга");

    if compare(num1, num2, num3) == 1 {
        println!("Equal");
    }
    else {
        println!("Not equal");
    }

}
